### PR TITLE
prelude: add protocol information through JSON

### PR DESCRIPTION
**Pull request by ToToL**
Prelude is based on what can be found in packet but now, Suricata provide more information in JSON format for HTTP, TLS, SSH and SMTP. This PR add the support for prelude to read these JSON and integrate them into IDMEF alerts.

### CODE REVIEW

File: src/alert-prelude.c [hunk 1]
```diff
@@ -478,6 +575,173 @@ static int PacketToDataV6(const Packet *p, const PacketAlert *pa, idmef_alert_t
...
+    const char *key_js;
+    json_t *js, *s_js, *value_js;
+    char local_key[128];
+
+    js = JsonHttpAddMetadata(p->flow, pa->tx_id);
+    if (js == NULL)
+        return;
+
+    json_object_foreach(s_js, key_js, value_js) {
+        snprintf(local_key, sizeof(char)*128, "http_%s", key_js);
```
Comment (inliniac) [inline src/alert-prelude.c]: sizeof(local_key) seems more robust

Comment (inliniac) [inline src/alert-prelude.c]: this pattern is in more places, can you replace them all?

**Comment by ToToL** [comment]
I updated my commit, is it now OK ?

**Comment by inliniac** [comment]
Please don't update existing PRs/branches: https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Github_work_flow#Create-a-new-branch-for-incorporating-feedback

**Comment by ToToL** [comment]
note that in the wiki, this command "git pull upstream --rebase master" give me "error: unknown option `rebase'"

**Pull request closed without merging**